tests/unit/charms/sdcore_nrf/v0/dummy_provider_charm/src/charm.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
```

Comment:
I'm a bit hastiest about having a dummy provider charm, since the `nrf` charm is already the provider side of this interface, so we could just implement it and test that directly.
But this is an opinion only, let me know if there is value from having the test charm separately.

Reply:
We want to test the interface, if something fails it will be more difficult to discern whether the issue comes from the NRF charm or from the interface itself. I do see the value in the **trust** we have in the unit tests, besides, maintaining the dummy charm it's pretty much effortless.

Of course, I think we must add the corresponding tests for the interface implementation in the NRF charm, for instance as we do [here](https://github.com/canonical/charmed-magma-orchestrator/blob/main/orc8r-nginx-operator/tests/unit/test_orchestator_relation.py) for the `orchestrator` relation in `orc8r-nginx` .

Let me know if that makes sense.